### DISCUSSION TITLE
Fix variables (No optimization over them)

### DISCUSSION THREAD

**Discussion by amir-hoseyn77** [osqp/osqp#590] [Mathematical Modeling]
Hi
Is there any way to fix (freeze) some variables (e.g. indices 0,1,2 of total var inds 0,1,2,3,4,5,6,7,8) with a fixed value? I mean some functions like `fix` in `pyomo`? For example in the below snippet by fixing x, the optimal solution changes.
```
import pyomo.environ as po
model = po.ConcreteModel()
model.x = po.Var(); model.y = po.Var();  # Var x, y
model.obj = po.Objective(sense=po.minimize, expr= model.x**2 + (model.x-model.y)**2)  # like rosenbrock
model.x.fix(1)  # fix `x` value to 1
# After Solving==> y=1 because x=1 fixed
```

**Comment by imciner2** [comment]
We don't provide a direct API to fix a variable, however you can always set the variable to have the same lower and upper bound to enforce an equality constraint to a fixed value on that variable.
